plugins/StereoControl/StereoControl.cpp

```diff
@@ -0,0 +1,443 @@
+	m_haasBuf[0].resize(m_haasBufSize);
+	m_haasBuf[1].resize(m_haasBufSize);
+}
+
+
+
+
+StereoControlEffect::~StereoControlEffect()
+{
+}
+
+
+
+
+bool StereoControlEffect::processAudioBuffer(sampleFrame* buf, const fpp_t frames)
+{
+	if(!isEnabled() || !isRunning ())
+	{
+		return false;
+	}
+
+	double outSum = 0.0;
+	const float d = dryLevel();
+	const float w = wetLevel();
+	
+	const ValueBuffer * gainBuf = m_stereocontrolControls.m_gainModel.valueBuffer();
+	const ValueBuffer * stereoizerBuf = m_stereocontrolControls.m_stereoizerModel.valueBuffer();
+	const ValueBuffer * widthBuf = m_stereocontrolControls.m_widthModel.valueBuffer();
+	const ValueBuffer * panModeBuf = m_stereocontrolControls.m_panModeModel.valueBuffer();
+	const ValueBuffer * panBuf = m_stereocontrolControls.m_panModel.valueBuffer();
+	const ValueBuffer * monoBuf = m_stereocontrolControls.m_monoModel.valueBuffer();
+	const ValueBuffer * dcBuf = m_stereocontrolControls.m_dcModel.valueBuffer();
+	const ValueBuffer * muteBuf = m_stereocontrolControls.m_muteModel.valueBuffer();
+	const ValueBuffer * monoBassBuf = m_stereocontrolControls.m_monoBassModel.valueBuffer();
+	const ValueBuffer * auditionBuf = m_stereocontrolControls.m_auditionModel.valueBuffer();
+	const ValueBuffer * invertLBuf = m_stereocontrolControls.m_invertLModel.valueBuffer();
+	const ValueBuffer * invertRBuf = m_stereocontrolControls.m_invertRModel.valueBuffer();
+	const ValueBuffer * soloChannelBuf = m_stereocontrolControls.m_soloChannelModel.valueBuffer();
+	const ValueBuffer * monoBassFreqBuf = m_stereocontrolControls.m_monoBassFreqModel.valueBuffer();
+	const ValueBuffer * stereoizerLPBuf = m_stereocontrolControls.m_stereoizerLPModel.valueBuffer();
+	const ValueBuffer * stereoizerHPBuf = m_stereocontrolControls.m_stereoizerHPModel.valueBuffer();
+	const ValueBuffer * panSpectralBuf = m_stereocontrolControls.m_panSpectralModel.valueBuffer();
+	const ValueBuffer * panDelayBuf = m_stereocontrolControls.m_panDelayModel.valueBuffer();
+	const ValueBuffer * panDualLBuf = m_stereocontrolControls.m_panDualLModel.valueBuffer();
+	const ValueBuffer * panDualRBuf = m_stereocontrolControls.m_panDualRModel.valueBuffer();
+
+	for(fpp_t f = 0; f < frames; ++f)
+	{
+		const float gain = dbfsToAmp(gainBuf ? gainBuf->value(f) : m_stereocontrolControls.m_gainModel.value());
+		const float stereoizer = (stereoizerBuf ? stereoizerBuf->value(f) : m_stereocontrolControls.m_stereoizerModel.value()) * 0.01f;
+		const float width = (widthBuf ? widthBuf->value(f) : m_stereocontrolControls.m_widthModel.value()) * 0.01f;
+		const int panMode = panModeBuf ? panModeBuf->value(f) : m_stereocontrolControls.m_panModeModel.value();
+		const float pan = (panBuf ? panBuf->value(f) : m_stereocontrolControls.m_panModel.value()) * 0.01f;
+		const bool mono = monoBuf ? monoBuf->value(f) : m_stereocontrolControls.m_monoModel.value();
+		const bool dc = dcBuf ? dcBuf->value(f) : m_stereocontrolControls.m_dcModel.value();
```

Comment:
It's hard to guess what `dc` means. Renaming to something like `dcRemoval` will make the meaning clear.
This also applies to related variables, too.